internal/btf/types.go

```diff
@@ -43,13 +43,24 @@ func (v *Void) size() uint32    { return 0 }
 func (v *Void) copy() Type      { return (*Void)(nil) }
 func (v *Void) walk(*copyStack) {}
 
+type IntEncoding byte
+
+const (
+	Signed IntEncoding = 1 << 0
+	Char   IntEncoding = 1 << 1
+	Bool   IntEncoding = 1 << 2
+)
+
 // Int is an integer of a given length.
 type Int struct {
 	TypeID
 	Name
 
 	// The size of the integer in bytes.
-	Size uint32
+	Size     uint32
+	Encoding IntEncoding
+	Offset   uint32
```

Comment:
What does `Offset` mean here?

Reply:
From https://www.kernel.org/doc/html/latest/bpf/btf.html
> The BTF_INT_OFFSET() specifies the starting bit offset to calculate values for this int. For example, a bitfield struct member has:

> btf member bit offset 100 from the start of the structure,
btf member pointing to an int type,
the int type has BTF_INT_OFFSET() = 2 and BTF_INT_BITS() = 4
Then in the struct memory layout, this member will occupy 4 bits starting from bits 100 + 2 = 102.

> Alternatively, the bitfield struct member can be the following to access the same bits as the above:

> btf member bit offset 102,
btf member pointing to an int type,
the int type has BTF_INT_OFFSET() = 0 and BTF_INT_BITS() = 4
The original intention of BTF_INT_OFFSET() is to provide flexibility of bitfield encoding. Currently, both llvm and pahole generate BTF_INT_OFFSET() = 0 for all int types.

Reply:
Thanks! Can you add a comment that Offset is in bits, not bytes?

Reply:
Added a comment.